spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -666,6 +667,34 @@ public static Dataset<Row> loadMetadataTable(
         spark, DataSourceV2Relation.create(metadataTable, Some.empty(), Some.empty(), options));
   }
 
+  public static String wapBranch(SparkSession spark, String branch) {
```

Comment:
I'm not sure it's clear enough from this name what this method is doing, can we add a java doc or possibly a new name?

Looks like it returns the WAP Branch if it is configured in the Spark conf and then the string that is passed in otherwise?

Reply:
HI Russell, I have added Javadoc for this method. Please review.

Reply:
I don't think this is much clearer. What I think this method is more likely defined as "determine write branch?" It is not restricted to WAP and also validates the configuration of non WAP config? It seems like it handles multiple concerns here

Reply:
Thanks @RussellSpitzer .
Would request your review once more. Made some changes in Javadoc

Reply:
Rather than having this function do both things we should probably have two functions.

1. Validate configuration
2. Choose branch

In general if our javadoc says a function is doing two different things, it means that function needs to be split. 

so code could look like
```java
deleteWhere() {
   validateConfig
   
   
   branch = branch != null ? branch : wapBranch
   if (branch != null) {
     deleteFiles.toBranch(branch)
   }
```